exporter/SynthesisFusionAddin/src/UI/ConfigCommand.py

```diff
@@ -1008,35 +1001,31 @@ def notify(self, args):
                 self.log.error("Could not execute configuration due to failure")
                 return
 
-            export_as_part_boolean = (
-                eventArgs.command.commandInputs.itemById("advanced_settings")
-                .children.itemById("exporter_settings")
-                .children.itemById("export_as_part")
-            ).value
-
             processedFileName = gm.app.activeDocument.name.replace(" ", "_")
             dropdownExportMode = INPUTS_ROOT.itemById("mode")
             if dropdownExportMode.selectedItem.index == 0:
                 isRobot = True
             elif dropdownExportMode.selectedItem.index == 1:
                 isRobot = False
 
-            if isRobot:
-                savepath = FileDialogConfig.SaveFileDialog(
-                    defaultPath=exporterOptions.fileLocation,
-                    ext="Synthesis File (*.synth)",
-                )
-            else:
-                savepath = FileDialogConfig.SaveFileDialog(defaultPath=exporterOptions.fileLocation)
-
-            if savepath == False:
-                # save was canceled
-                return
+            if exporterOptions.exportLocation == ExportLocation.DOWNLOAD:
+                if isRobot:
+                    savepath = FileDialogConfig.SaveFileDialog(
+                        defaultPath=exporterOptions.fileLocation,
+                        ext="Synthesis File (*.synth)",
+                    )
+                else:
+                    savepath = FileDialogConfig.SaveFileDialog(defaultPath=exporterOptions.fileLocation)
 
-            updatedPath = pathlib.Path(savepath).parent
-            if updatedPath != self.current.filePath:
-                self.current.filePath = str(updatedPath)
+                if savepath == False:
+                    # save was canceled
+                    return
 
+                updatedPath = pathlib.Path(savepath).parent
+                if updatedPath != self.current.filePath:
+                    self.current.filePath = str(updatedPath)
+            else:
+                savepath = processedFileName
```

Comment:
This `if` statement should be using the current selected export location not the one currently saved to the design. Which is what the `exporterOptions` object is here.